src/Dictionaries/ComplexKeyHashedDictionary.cpp

```diff
@@ -560,7 +497,18 @@ void ComplexKeyHashedDictionary::getItemsImpl(
         const auto key = placeKeysInPool(i, key_columns, keys, temporary_keys_pool);
 
         const auto it = attr.find(key);
-        set_value(i, it ? static_cast<OutputType>(it->getMapped()) : get_default(i));
+
+        if (it)
+        {
+            set_value(i, static_cast<OutputType>(it->getMapped()), false);
+        }
+        else
+        {
+            if (attribute.is_nullable && attribute.nullable_set->find(key) != nullptr)
```

Comment:
Weird logic. How we update nullable_set?

Reply:
In ComplexKeyHashedDictionary::setAttributeValue.